docs/guides/src/main/server/fips.adoc

```diff
@@ -0,0 +1,229 @@
+
+* `bc-fips-1.0.2.3.jar`
+* `bctls-fips-1.0.14.jar`
+* `bcpkix-fips-1.0.7.jar`
+
+== Generating keystore
+
+You can create either `pkcs12` or `bcfks` keystore to be used for the Keycloak server SSL. The `pkcs12` works well in BCFIPS non-approved mode.
+
+PKCS12 keystore can be generated with OpenJDK 17 Java on RHEL 9 in the standard way.
+
+BCFKS keystore generation requires the use of the BouncyCastle FIPS libraries and a custom security file.
+
+You can start with create a helper file, such as `/tmp/kc.keystore-create.java.security`. The content of the file needs only to have the following property:
+```
+securerandom.strongAlgorithms=PKCS11:SunPKCS11-NSS-FIPS
+```
+
+Next enter a command such as the following to generate the keystore:
+```
+keytool -keystore $KEYCLOAK_HOME/conf/server.keystore \
+  -storetype bcfks \
+  -providername BCFIPS \
+  -providerclass org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider \
+  -provider org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider \
+  -providerpath $KEYCLOAK_HOME/providers/bc-fips-*.jar \
+  -alias localhost \
+  -genkeypair -sigalg SHA512withRSA -keyalg RSA -storepass passwordpassword \
+  -dname CN=localhost -keypass passwordpassword \
+  -J-Djava.security.properties=/tmp/kc.keystore-create.java.security
+```
+
+WARNING: Using self-signed certificates is for demonstration purposes only, so replace these certificates with proper certificates when you move to a production environment.
+
+Similar options are needed when you are doing any other manipulation with keystore/truststore of `bcfks` type.
+
+== Running the server.
+
+To run the server with BCFIPS in non-approved mode, enter the following command::
+
+<@kc.start parameters="--fips-mode=enabled --hostname=localhost --https-key-store-password=passwordpassword --log-level=INFO,org.keycloak.common.crypto:TRACE,org.keycloak.crypto:TRACE"/>
+
+NOTE: You can disable logging in production if everything works as expected.
+
+== Strict mode
+
+For the `fips-mode` option, the more secure alternative is to use `--fips-mode=strict` in which case BouncyCastle FIPS will use "approved mode".
+Using that option results in stricter security requirements on cryptography and security algorithms.
+
+When starting server, you can check that the startup log contains `KC` provider with the note about `Approved Mode` such as the following:
+
+```
+KC(BCFIPS version 1.000203 Approved Mode, FIPS-JVM: enabled) version 1.0 - class org.keycloak.crypto.fips.KeycloakFipsSecurityProvider,
+```
+
+=== Cryptography restrictions in strict mode
+
+* As mentioned in the previous section, strict mode may not work with `pkcs12` keystore. It is needed to use another keystore (like `bcfks`) as mentioned earlier. Also `jks` and `pkcs12` keystores are not
+supported in Keycloak. Some examples are importing or generating a keystore of an OIDC or SAML client in the Admin Console or for a `java-keystore` provider in the realm keys.
```

Comment:
Why pkcs12 is not supported in keyloak, you mean in strict mode? Right? If I'm right I would add `Also jks and pkcs12 keystores are not supported in Keycloak when using strict mode`.